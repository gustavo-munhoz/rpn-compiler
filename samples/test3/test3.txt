(1024 512 -)
(0.125 8 *)
((1 RES) (2 RES) +)
((3 RES) MEM)
(16 0.5 |)
((MEM) (1 RES) /)
((-1.5 4 ^) (2 RES) +)
(-80.5 7 %)
((6 RES) (8 RES) -)
((1 RES) MEM)
((7 RES) 0.125 +)
(MEM)
(((1 RES) (2 RES) +) (3 RES) *)
(0.125 1000 |)
(37.5 -7 %)
(4 IF)
((MEM) THEN)
((2 RES) ELSE)
(6 2 FOR)
