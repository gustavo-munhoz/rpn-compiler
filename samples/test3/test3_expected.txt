<L_PAREN '(' @1:1>
<NUM_INT '1024' @1:2>
<NUM_INT '512' @1:7>
<ARITHMETIC_OP '-' @1:11>
<R_PAREN ')' @1:12>
<L_PAREN '(' @2:1>
<NUM_FLOAT '0.125' @2:2>
<NUM_INT '8' @2:8>
<ARITHMETIC_OP '*' @2:10>
<R_PAREN ')' @2:11>
<L_PAREN '(' @3:1>
<L_PAREN '(' @3:2>
<NUM_INT '1' @3:3>
<KEYWORD 'RES' @3:5>
<R_PAREN ')' @3:8>
<L_PAREN '(' @3:10>
<NUM_INT '2' @3:11>
<KEYWORD 'RES' @3:13>
<R_PAREN ')' @3:16>
<ARITHMETIC_OP '+' @3:18>
<R_PAREN ')' @3:19>
<L_PAREN '(' @4:1>
<L_PAREN '(' @4:2>
<NUM_INT '3' @4:3>
<KEYWORD 'RES' @4:5>
<R_PAREN ')' @4:8>
<KEYWORD 'MEM' @4:10>
<R_PAREN ')' @4:13>
<L_PAREN '(' @5:1>
<NUM_INT '16' @5:2>
<NUM_FLOAT '0.5' @5:5>
<ARITHMETIC_OP '|' @5:9>
<R_PAREN ')' @5:10>
<L_PAREN '(' @6:1>
<L_PAREN '(' @6:2>
<KEYWORD 'MEM' @6:3>
<R_PAREN ')' @6:6>
<L_PAREN '(' @6:8>
<NUM_INT '1' @6:9>
<KEYWORD 'RES' @6:11>
<R_PAREN ')' @6:14>
<ARITHMETIC_OP '/' @6:16>
<R_PAREN ')' @6:17>
<L_PAREN '(' @7:1>
<L_PAREN '(' @7:2>
<NUM_FLOAT '-1.5' @7:3>
<NUM_INT '4' @7:8>
<ARITHMETIC_OP '^' @7:10>
<R_PAREN ')' @7:11>
<L_PAREN '(' @7:13>
<NUM_INT '2' @7:14>
<KEYWORD 'RES' @7:16>
<R_PAREN ')' @7:19>
<ARITHMETIC_OP '+' @7:21>
<R_PAREN ')' @7:22>
<L_PAREN '(' @8:1>
<NUM_FLOAT '-80.5' @8:2>
<NUM_INT '7' @8:8>
<ARITHMETIC_OP '%' @8:10>
<R_PAREN ')' @8:11>
<L_PAREN '(' @9:1>
<L_PAREN '(' @9:2>
<NUM_INT '6' @9:3>
<KEYWORD 'RES' @9:5>
<R_PAREN ')' @9:8>
<L_PAREN '(' @9:10>
<NUM_INT '8' @9:11>
<KEYWORD 'RES' @9:13>
<R_PAREN ')' @9:16>
<ARITHMETIC_OP '-' @9:18>
<R_PAREN ')' @9:19>
<L_PAREN '(' @10:1>
<L_PAREN '(' @10:2>
<NUM_INT '1' @10:3>
<KEYWORD 'RES' @10:5>
<R_PAREN ')' @10:8>
<KEYWORD 'MEM' @10:10>
<R_PAREN ')' @10:13>
<L_PAREN '(' @11:1>
<L_PAREN '(' @11:2>
<NUM_INT '7' @11:3>
<KEYWORD 'RES' @11:5>
<R_PAREN ')' @11:8>
<NUM_FLOAT '0.125' @11:10>
<ARITHMETIC_OP '+' @11:16>
<R_PAREN ')' @11:17>
<L_PAREN '(' @12:1>
<KEYWORD 'MEM' @12:2>
<R_PAREN ')' @12:5>
<L_PAREN '(' @13:1>
<L_PAREN '(' @13:2>
<L_PAREN '(' @13:3>
<NUM_INT '1' @13:4>
<KEYWORD 'RES' @13:6>
<R_PAREN ')' @13:9>
<L_PAREN '(' @13:11>
<NUM_INT '2' @13:12>
<KEYWORD 'RES' @13:14>
<R_PAREN ')' @13:17>
<ARITHMETIC_OP '+' @13:19>
<R_PAREN ')' @13:20>
<L_PAREN '(' @13:22>
<NUM_INT '3' @13:23>
<KEYWORD 'RES' @13:25>
<R_PAREN ')' @13:28>
<ARITHMETIC_OP '*' @13:30>
<R_PAREN ')' @13:31>
<L_PAREN '(' @14:1>
<NUM_FLOAT '0.125' @14:2>
<NUM_INT '1000' @14:8>
<ARITHMETIC_OP '|' @14:13>
<R_PAREN ')' @14:14>
<L_PAREN '(' @15:1>
<NUM_FLOAT '37.5' @15:2>
<NUM_INT '-7' @15:7>
<ARITHMETIC_OP '%' @15:10>
<R_PAREN ')' @15:11>
<L_PAREN '(' @16:1>
<NUM_INT '4' @16:2>
<KEYWORD 'IF' @16:4>
<R_PAREN ')' @16:6>
<L_PAREN '(' @17:1>
<L_PAREN '(' @17:2>
<KEYWORD 'MEM' @17:3>
<R_PAREN ')' @17:6>
<KEYWORD 'THEN' @17:8>
<R_PAREN ')' @17:12>
<L_PAREN '(' @18:1>
<L_PAREN '(' @18:2>
<NUM_INT '2' @18:3>
<KEYWORD 'RES' @18:5>
<R_PAREN ')' @18:8>
<KEYWORD 'ELSE' @18:10>
<R_PAREN ')' @18:14>
<L_PAREN '(' @19:1>
<NUM_INT '6' @19:2>
<NUM_INT '2' @19:4>
<KEYWORD 'FOR' @19:6>
<R_PAREN ')' @19:9>
